backend/pkg/utils/utils.go

```diff
@@ -26,3 +27,27 @@ func DedupeSliceOrdered[T comparable](input []T) []T {
 	}
 	return output
 }
+
+func DedupeSlice[T comparable](input []T) []T {
+	set := map[T]any{}
+	for _, i := range input {
+		set[i] = struct{}{}
+	}
+	output := make([]T, 0, len(set))
+	for key := range set {
+		output = append(output, key)
+	}
+	return output
+}
+
+func CompareSlices(slice1, slice2 []string) bool {
```

Comment:
I think we can remove this and use [slices.Compare](https://pkg.go.dev/slices#Compare)